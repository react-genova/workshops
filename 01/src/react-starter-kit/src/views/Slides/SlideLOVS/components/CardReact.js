//@flow
import React from 'react';
import { Card, CardElement } from '../../../Card';

type Props = {
    onClick:(url:string)=>void
};

const CardReact = ({onClick}:Props) => (
    <Card title="React" description="here is the big guy">
        <CardElement title="create-react-app" description="the tool which simplifies the mess" url="https://github.com/facebookincubator/create-react-app/" onClick={onClick} />
        <CardElement title="recompose" description="HOC based useful library" url="https://github.com/acdlite/recompose/" onClick={onClick} />
        <CardElement title="styled-components" description="CSS inline to the next level" url="https://www.styled-components.com/" onClick={onClick} />
    </Card>
);

export default CardReact;