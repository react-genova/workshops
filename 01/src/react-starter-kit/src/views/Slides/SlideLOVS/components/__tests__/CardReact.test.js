//@flow
import React from 'react';
import { shallow } from 'enzyme';
import CardReact from '../CardReact';

describe('Card React', () => {

    it('snapshot check', () => {
        const onClick = (url:string)=>{url.toLocaleLowerCase();};
        const component = shallow(<CardReact onClick={onClick} />);
        expect(component).toMatchSnapshot();
    });

});