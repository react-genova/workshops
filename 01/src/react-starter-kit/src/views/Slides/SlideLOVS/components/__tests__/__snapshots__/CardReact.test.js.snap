// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card React snapshot check 1`] = `
<Card
  description="here is the big guy"
  title="React"
>
  <CardElement
    description="the tool which simplifies the mess"
    onClick={[Function]}
    title="create-react-app"
    url="https://github.com/facebookincubator/create-react-app/"
  />
  <CardElement
    description="HOC based useful library"
    onClick={[Function]}
    title="recompose"
    url="https://github.com/acdlite/recompose/"
  />
  <CardElement
    description="CSS inline to the next level"
    onClick={[Function]}
    title="styled-components"
    url="https://www.styled-components.com/"
  />
</Card>
`;
